python/tank/util/metrics.py

```diff
@@ -64,15 +68,33 @@ def __new__(cls, *args, **kwargs):
 
         return cls.__instance
 
-    def log(self, metric):
+    def log(self, metric, log_once=False):
         """Add the metric to the queue for dispatching.
 
-        :param ToolkitMetric metric: The metric to log.
+        If ``log_once`` is set to ``True``, this will only log the metric if it
+        is the first attempt to log it.
 
+        :param ToolkitMetric metric: The metric to log.
+        :param bool log_once: ``True`` if this metric should be ignored if it
+            has already been logged. ``False`` otherwise. Defaults to ``False``.
         """
+
+        # This assumes that supplied object's classes implement __repr__
+        # to return consistent results when building objects with the same
+        # internal data. See the UserActivityMetric and UserAttributeMetric
+        # classes below.
+        metric_identifier = repr(metric)
+
+        if log_once and metric_identifier in self.__logged_metrics:
+            # the metric is already logged! nothing to do.
+            return
+
         self._lock.acquire()
         try:
             self._queue.append(metric)
+
+            # remember that we've logged this one already
+            self.__logged_metrics.add(metric_identifier)
```

Comment:
My only thought is that maybe there's code logging the same metric in a couple of places... maybe? And one of those only wants to log the metric if it hasn't already been logged... maybe? 

These are just short strings, and we're using a `set`, so i don't think this will get too big within a single session. Let me know if you disagree. Maybe ok to leave this for now?